src/vm/eventpipeeventinstance.cpp

```diff
@@ -53,7 +55,7 @@ EventPipeEventInstance::EventPipeEventInstance(
     m_dataLength = length;
     QueryPerformanceCounter(&m_timeStamp);
 
-    if(event.NeedStack())
+    if(event.NeedStack() && !session.RundownEnabled())
```

Comment:
The event objects we see during rundown should be different than the ones we see otherwise (right?). Instead of worrying out whether we are in rundown or not here, could we instead set m_needStack to false when creating the event in InitDotNETRuntimeRundown (generated code) ?

Reply:
Right now, the set of events that gets emitted during rundown is from the rundown provider and the public provider (Microsoft-Windows-DotNETRuntime).  Fundamentally, needstack should actually come from the session rather than from the provider because it is up to the consumer to state what data it wants.  So, we should actually remove needstack.

Reply:
OK, that makes sense. So then this is a step in that direction